source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -284,29 +289,57 @@ FaultFilter::delayDuration(const Http::RequestHeaderMap& request_headers) {
   return ret;
 }
 
-absl::optional<Http::Code>
-FaultFilter::abortHttpStatus(const Http::RequestHeaderMap& request_headers) {
+AbortHttpAndGrpcStatus FaultFilter::abortStatus(const Http::RequestHeaderMap& request_headers) {
   if (!isAbortEnabled(request_headers)) {
-    return absl::nullopt;
+    return std::make_pair(absl::nullopt, absl::nullopt);
   }
 
+  auto grpc_status = abortGrpcStatus(request_headers);
+
+  // If gRPC status code is set, then HTTP will be set to 200
+  return grpc_status.has_value() ? std::make_pair(Http::Code::OK, grpc_status)
+                                 : std::make_pair(abortHttpStatus(request_headers), grpc_status);
+}
+
+absl::optional<Http::Code>
+FaultFilter::abortHttpStatus(const Http::RequestHeaderMap& request_headers) {
   // See if the configured abort provider has a default status code, if not there is no abort status
   // code (e.g., header configuration and no/invalid header).
-  const auto config_abort = fault_settings_->requestAbort()->statusCode(&request_headers);
-  if (!config_abort.has_value()) {
+  auto http_status = fault_settings_->requestAbort()->httpStatusCode(
+      request_headers.get(Filters::Common::Fault::HeaderNames::get().AbortRequest));
+  if (!http_status.has_value()) {
     return absl::nullopt;
   }
 
-  auto status_code = static_cast<uint64_t>(config_abort.value());
-  auto code = static_cast<Http::Code>(config_->runtime().snapshot().getInteger(
-      fault_settings_->abortHttpStatusRuntime(), status_code));
+  auto default_http_status_code = static_cast<uint64_t>(http_status.value());
+  auto runtime_http_status_code = static_cast<Http::Code>(config_->runtime().snapshot().getInteger(
+      fault_settings_->abortHttpStatusRuntime(), default_http_status_code));
 
   if (!downstream_cluster_abort_http_status_key_.empty()) {
-    code = static_cast<Http::Code>(config_->runtime().snapshot().getInteger(
-        downstream_cluster_abort_http_status_key_, status_code));
+    runtime_http_status_code = static_cast<Http::Code>(config_->runtime().snapshot().getInteger(
+        downstream_cluster_abort_http_status_key_, default_http_status_code));
   }
 
-  return code;
+  return runtime_http_status_code;
+}
+
+absl::optional<Grpc::Status::GrpcStatus>
+FaultFilter::abortGrpcStatus(const Http::RequestHeaderMap& request_headers) {
+  auto grpc_status = fault_settings_->requestAbort()->grpcStatusCode(
+      request_headers.get(Filters::Common::Fault::HeaderNames::get().AbortGrpcRequest));
+
+  auto default_grpc_status_code = grpc_status.has_value()
```

Comment:
shouldn't we return an `absl::nullptr` from here if `grpc_status` doesn't have a value? We do a similar thing in the implementation of `abortStatus` method.

Reply:
intention here is to support setting gRPC status even through just runtime config. For example, even if abort fault filter configuration and header doesn't have grpc_status but if the abort_grpc_status_runtime runtime is set we still want to set the grpc_status. Does this make sense ?

Reply:
The use case you want to support definitely makes sense. 👍 

I'm just a little bit worried that GRPC aborts will end up using proto config in a way that's different from the way other faults use their protos. "Fixed" fault provides - including HTTP status code, response rate limit and delay request - all require proto config values to be present. For the user of the Envoy, it may be a little bit confusing if for GRPC aborts we follow a different convention. 

That being said, if runtime-only config is the thing you want to support I'm going to leave it up to more Envoy experienced folks to decide whether it's ok or no to have this fault provider not require full proto config (no GRPC status code required in proto config).

Reply:
I agree with @Augustyniak that we should be consistent here. I'm going to defer to @Augustyniak for initial review of this PR and I will do a pass after. :)